[MCAPI-52] fix: move upgrade cluster to integration

ricolin opened the pull request:
fixed https://github.com/vexxhost/magnum-cluster-api/issues/451

ricolin committed:
CI: skip upgrade with kube_tag in functional test

ricolin committed:
Add upgrade tests to integration test

ricolin committed:
tunning integration for upgrade

ricolin committed:
Add Tempest support

ricolin committed:
fix tempest

okozachenko1203 commented on magnum_cluster_api/tests/functional/conftest.py:
```diff
@@ -25,6 +25,7 @@ def cluster(
     mock_validate_cluster,
     mock_osc,
     mock_certificates,
+    mock_cache,
```
```suggestion
    mock_get_server_group,
```

okozachenko1203 commented on magnum_cluster_api/tests/conftest.py:
```diff
@@ -273,3 +279,11 @@ def server_side_apply_matcher():
     )
+
+
+@pytest.fixture(scope="session")
+def mock_cache(session_mocker):
+    session_mocker.patch(
+        "magnum_cluster_api.cache.ServerGroupCache.get", return_value=None
+    )
+    session_mocker.patch("magnum_cluster_api.cache.ServerGroupCache.set")
```
```suggestion

```

okozachenko1203 commented on magnum_cluster_api/tests/conftest.py:
```diff
@@ -95,6 +95,12 @@ def mock_osc(session_mocker, image):
         )
     )
 
+    # Nova
+
+    mock_nova_client = mock_clients.nova.return_value
+    mock_nova_client.server_groups.create.return_value = (
+        openstack.compute.v2.server_group.ServerGroup(id=uuidutils.generate_uuid())
+    )
```
```suggestion
```

ricolin committed:
Add mock_get_server_group to functional

ricolin committed:
fix tests

okozachenko1203 commented:
@ricolin https://github.com/openstack/magnum-tempest-plugin/commit/22272488bc746d1c8eb8a1f8a979ad5af8dc4457

ricolin committed:
targeting docker_storage_driver config limit

https://github.com/openstack/magnum-tempest-plugin/commit/22272488bc746d1c8eb8a1f8a979ad5af8dc4457

ricolin commented:
> @ricolin [openstack/magnum-tempest-plugin@2227248](https://github.com/openstack/magnum-tempest-plugin/commit/22272488bc746d1c8eb8a1f8a979ad5af8dc4457)

hum, no overlay2 ; |

ricolin committed:
chores: clean side note

ricolin committed:
test: Enable k8s access test + tempest

ricolin committed:
allow newest to run accessible check

ricolin committed:
correct count on nodes

ricolin committed:
rename zuul jobs

okozachenko1203 commented:
recheck

ricolin commented:
Okay, somehow we hit a magnum bug: https://review.opendev.org/c/openstack/magnum/+/940815
